tests/console/libgcrypt.pm

```diff
@@ -0,0 +1,55 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: test that libgcrypt works as intended by executing
+#          the selftest and by calling libgcrypt-config.
+#          In addition to this test, also the gpg test should be
+#          checked by tester because gpg uses libgcrypt.
+# - execute libgcrypt self test
+# - using libgcrypt-config, check some parameters
+# Maintainer: Paolo Stivanin <pstivanin@suse.com>
+
+use base "opensusebasetest";
+use testapi;
+use utils;
+use strict;
+use warnings;
+use version_utils qw(is_sle is_opensuse);
+use registration;
+
+sub run {
+    select_console 'root-console';
+
+    assert_script_run "rpm -q libgcrypt20";
+    if (script_run("rpm -q libgcrypt-devel") == 1) {
+        # sle-sdk can be added only for 12.3 and 12.4. On 15.x, the package is part of basesystem
+        if (is_sle("<15")) {
+            register_product;
```

Comment:
@mimi1vx @baierjan what is our current stance on adding products/addons via registering it? IIRC there was decision to go with adding repo from OBS to avoid possible network issues. also, there is no de-registering afterwards, so depending on placement it could interfere with later tests

Reply:
@Hoorhay you mean doing a `zypper ar http://download.suse.de/ibs/SUSE/Updates/SLE-SDK/${version}/${arch}/update/ sdk-tmp` and then ref and install?

Reply:
@paolostivanin yes, that was my idea. but I don't want to force you to change it on the basis of my speculations, let's hear from Jan and/or Ondrej

Reply:
oki! Thanks!

Reply:
I am not sure about the registration, is there a reason for it? We can schedule the test somewhere with SDK included in `SCC_ADDONS`, can't we?

Reply:
@baierjan the only reason is that the test needs `libgcrypt-devel` which is present in the sdk repo in SLES 12*.
Yes, the alternative would be to remove the registration part and schedule it where the sdk is already available, good point :+1:

Reply:
I didn't find an existing test suite with `sdk` in `SCC_ADDONS` :thinking: shall we go with @Hoorhay idea or should we create a new/change an existing test suite? @baierjan

Reply:
AFAIK  *mau_extratests* are installed with `sdk` (on 12-SP4+). Shouldn't that be enough?

Reply:
true, but what about 12.3 @baierjan ? :thinking:

Reply:
12.3 is in LTSS mode, there is no `sdk` available.

Reply:
right :)